### ISSUE TITLE
Failed to activate the save-session package

### ISSUE THREAD

**Issue by hailfire113** [mpeterson2/save-session#84]
[Enter steps to reproduce below:]

1.Close atom
2. Reopen atom

**Atom Version**: 0.187.0
**System**: Microsoft Windows 8.1 Pro
**Thrown From**: [save-session](https://github.com/mpeterson2/save-session) package, v0.12.1

### Stack Trace

Failed to activate the save-session package

```
At Cannot read property 'path' of undefined

TypeError: Cannot read property 'path' of undefined
    at Object.module.exports.saveFile (C:\Users\Alan\.atom\packages\save-session\lib\config.coffee:86:102)
    at Object.module.exports.activate (C:\Users\Alan\.atom\packages\save-session\lib\files.coffee:14:24)
    at Object.module.exports.activate (C:\Users\Alan\.atom\packages\save-session\lib\save-session.coffee:120:20)
    at Package.module.exports.Package.activateNow (C:\Users\Alan\AppData\Local\atom\app-0.187.0\resources\app\src\package.js:240:19)
    at C:\Users\Alan\AppData\Local\atom\app-0.187.0\resources\app\src\package.js:221:30
    at Package.module.exports.Package.measure (C:\Users\Alan\AppData\Local\atom\app-0.187.0\resources\app\src\package.js:167:15)
    at Package.module.exports.Package.activate (C:\Users\Alan\AppData\Local\atom\app-0.187.0\resources\app\src\package.js:214:14)
    at PackageManager.module.exports.PackageManager.activatePackage (C:\Users\Alan\AppData\Local\atom\app-0.187.0\resources\app\src\package-manager.js:465:21)
    at C:\Users\Alan\AppData\Local\atom\app-0.187.0\resources\app\src\package-manager.js:446:29
    at Config.module.exports.Config.transact (C:\Users\Alan\AppData\Local\atom\app-0.187.0\resources\app\src\config.js:350:16)
```

### Commands

```
```

### Config

```json
{
  "core": {
    "themes": [
      "one-light-ui",
      "one-light-syntax"
    ],
    "ignoredNames": [
      ".git"
    ]
  },
  "save-session": {
    "dataSaveFolder": "C:\\Users\\Alan\\.atom\\packages\\save-session\\projects"
  }
}
```

### Installed Packages

```coffee
# User
color-picker, v1.4.4
linter, v0.12.0
save-session, v0.12.1

# Dev
No dev packages
```

**Comment by mpeterson2** [comment]
0.12.2 should fix this issue. Thanks for reporting it.

**Comment by mrbirne** [comment]
I had the same problmen. I installed it over the Settings -> install -> save-session

**Atom Version**: 0.187.0
**System**: linux 3.13.0-46-generic
**Thrown From**: [save-session](https://github.com/mpeterson2/save-session) package, v0.12.2

### Stack Trace

Failed to activate the save-session package

```
At Cannot read property 'path' of undefined

TypeError: Cannot read property 'path' of undefined
    at Object.module.exports.saveFile (/home/maurice/.atom/packages/save-session/lib/config.coffee:86:112)
    at Object.module.exports.activate (/home/maurice/.atom/packages/save-session/lib/files.coffee:14:24)
    at Object.module.exports.activate (/home/maurice/.atom/packages/save-session/lib/save-session.coffee:120:20)
    at Package.module.exports.Package.activateNow (/usr/share/atom/resources/app/src/package.js:240:19)
    at /usr/share/atom/resources/app/src/package.js:221:30
    at Package.module.exports.Package.measure (/usr/share/atom/resources/app/src/package.js:167:15)
    at Package.module.exports.Package.activate (/usr/share/atom/resources/app/src/package.js:214:14)
    at PackageManager.module.exports.PackageManager.activatePackage (/usr/share/atom/resources/app/src/package-manager.js:465:21)
    at /usr/share/atom/resources/app/src/package-manager.js:446:29
    at Config.module.exports.Config.transact (/usr/share/atom/resources/app/src/config.js:350:16)
```

### Commands

```
```

### Config

```json
{
  "core": {
    "themes": [
      "atom-dark-ui",
      "atom-dark-syntax"
    ]
  },
  "save-session": {
    "dataSaveFolder": "/home/maurice/.atom/packages/save-session/projects"
  }
}
```

### Installed Packages

```coffee
# User
save-session, v0.12.2
terminal-panel, v1.10.0

# Dev
No dev packages
```

**Comment by tedyun** [comment]
Same error with save-session ver 0.12.2, atom ver 0.187, Windows 7. I manually created save-session/projects folder which fixed the problem.

**Comment by const314** [comment]
The same problem with v0.12.2 on OS X:
**Atom Version**: 0.187.0
**System**: Mac OS X 10.10.2
**Thrown From**: [save-session](https://github.com/mpeterson2/save-session) package, v0.12.2

**Comment by mrbirne** [comment]
May you have the wrong excess rights for a atom on the project folder.

**Comment by nklmilojevic** [comment]
The same thing happens to me in Windows 8.1 save-session/projects already exist.

**Comment by anatagawa** [comment]
do not use the package "open-last-project" , incompatiblity with this

**Comment by nklmilojevic** [comment]
I'm not using that package.

**Comment by mpeterson2** [comment]
Looks like this is still an issue then. I'll look into it again.

**Comment by mpeterson2** [comment]
Okay, this should be good now in 0.12.3. Let me know if there are still issues.

**Comment by nklmilojevic** [comment]
It works, thanks!

**Comment by tumri** [comment]
I got this error today as well.


[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 0.190.0
**System**: Microsoft Windows 7 Home Premium 
**Thrown From**: [save-session](https://github.com/mpeterson2/save-session) package, v0.13.1

### Stack Trace

Failed to activate the save-session package

```
At Cannot read property 'length' of undefined

TypeError: Cannot read property 'length' of undefined
  at Object.module.exports.saveFile (C:\Users\Tumri\.atom\packages\save-session\lib\config.coffee:97:44)
  at Object.module.exports.activate (C:\Users\Tumri\.atom\packages\save-session\lib\files.coffee:9:27)
  at Object.module.exports.activate (C:\Users\Tumri\.atom\packages\save-session\lib\save-session.coffee:100:11)
  at Package.module.exports.Package.activateNow (C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\src\package.js:225:19)
  at C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\src\package.js:206:30
  at Package.module.exports.Package.measure (C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\src\package.js:152:15)
  at Package.module.exports.Package.activate (C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\src\package.js:199:14)
  at PackageManager.module.exports.PackageManager.activatePackage (C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\src\package-manager.js:435:21)
  at C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\node_modules\settings-view\lib\package-manager.js:268:29
  at exit (C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\node_modules\settings-view\lib\package-manager.js:43:16)
  at BufferedProcess.triggerExitCallback (C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\src\buffered-process.js:63:47)
  at ChildProcess.<anonymous> (C:\Users\Tumri\AppData\Local\atom\app-0.190.0\resources\app\src\buffered-process.js:85:18)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:169:7)
  at Process.ChildProcess._handle.onexit (child_process.js:1057:12)

```

### Commands

```
     -6:01.3 application:add-project-folder (atom-text-editor.editor)
     -2:18.1 grammar-selector:show (atom-text-editor.editor)
     -1:09.1 deprecation-cop:view (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -0:36.4 core:confirm (atom-text-editor.editor.mini)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "linter-htmlhint"
    ]
  },
  "save-session": {
    "dataSaveFolder": "C:\\Users\\Tumri\\.atom\\packages\\save-session\\projects"
  }
}
```

### Installed Packages

```coffee
# User
color-picker, v1.6.0
linter, v0.12.0
linter-bootlint, v0.0.5
linter-clang, v2.25.1
linter-csslint, v0.0.11
linter-flexpmd, v0.1.9
linter-haml, v0.1.1
linter-hlint, v0.3.1
linter-javac, v0.1.3
linter-js-yaml, v1.0.0
linter-jscs, v1.10.0
linter-jshint, v0.1.2
linter-jsxhint, v0.1.1
linter-less, v0.3.1
linter-lintr, v0.0.1
linter-lua, v0.1.5
linter-php, v0.0.12
linter-phpcs, v0.0.12
linter-phpmd, v0.0.10
linter-pylint, v0.2.1
linter-rubocop, v0.2.2
linter-ruby, v0.1.4
linter-rust, v0.0.8
linter-scalac, v0.3.3
linter-scspell, v1.1.1
linter-scss-lint, v0.0.14
linter-shellcheck, v0.0.7
linter-stylint, v0.4.0
linter-tidy, v1.0.0
linter-tslint, v0.2.1
linter-xmllint, v0.0.5
minimap, v4.7.6
save-session, v0.13.1

# Dev
No dev packages
```

**Comment by Leonick** [comment]
Not getting the error every time I open Atom, only got it the first one or two times. Obviously isn't working properly though. Only open the tab that was active in last session, everything else is lost.

**Atom Version**: 0.189.0
**System**: Mac OS X 10.10.3
**Thrown From**: [save-session](https://github.com/mpeterson2/save-session) package, v0.13.1

### Stack Trace

Failed to activate the save-session package

```
At Cannot read property 'length' of undefined

TypeError: Cannot read property 'length' of undefined
  at Object.module.exports.saveFile (/Users/hampus/.atom/packages/save-session/lib/config.coffee:97:44)
  at Object.module.exports.activate (/Users/hampus/.atom/packages/save-session/lib/files.coffee:9:27)
  at Object.module.exports.activate (/Users/hampus/.atom/packages/save-session/lib/save-session.coffee:100:11)
  at Package.module.exports.Package.activateNow (/Users/hampus/Downloads/Atom.app/Contents/Resources/app/src/package.js:240:19)
  at /Users/hampus/Downloads/Atom.app/Contents/Resources/app/src/package.js:221:30
  at Package.module.exports.Package.measure (/Users/hampus/Downloads/Atom.app/Contents/Resources/app/src/package.js:167:15)
  at Package.module.exports.Package.activate (/Users/hampus/Downloads/Atom.app/Contents/Resources/app/src/package.js:214:14)
  at PackageManager.module.exports.PackageManager.activatePackage (/Users/hampus/Downloads/Atom.app/Contents/Resources/app/src/package-manager.js:461:21)
  at /Users/hampus/Downloads/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/package-manager.js:268:29
  at exit (/Users/hampus/Downloads/Atom.app/Contents/Resources/app/node_modules/settings-view/lib/package-manager.js:43:16)
  at BufferedProcess.triggerExitCallback (/Users/hampus/Downloads/Atom.app/Contents/Resources/app/src/buffered-process.js:63:47)
  at ChildProcess.<anonymous> (/Users/hampus/Downloads/Atom.app/Contents/Resources/app/src/buffered-process.js:85:18)
  at emitTwo (events.js:87:13)
  at ChildProcess.emit (events.js:169:7)
  at Process.ChildProcess._handle.onexit (child_process.js:1057:12)

```

### Commands

```
  3x -1:55.2 window:reset-font-size (atom-pane.pane.active)
  4x -1:53.5 tree-view:toggle-focus (atom-pane.pane.active)
  3x -1:50.3 core:move-down (atom-pane.pane.active)
     -1:49.1 tree-view:toggle-focus (atom-pane.pane.active)
  5x -1:48.5 core:move-down (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
  4x -1:47.6 core:move-up (ol.tree-view.full-menu.list-tree.has-collapsable-children.focusable-panel)
     -1:31.7 core:confirm (atom-text-editor.editor.mini)
```

### Config

```json
{
  "core": {
    "themes": [
      "one-dark-ui",
      "one-dark-syntax"
    ],
    "disabledPackages": [
      "tabs",
      "tree-view"
    ]
  },
  "save-session": {
    "dataSaveFolder": "/Users/hampus/.atom/packages/save-session/projects"
  }
}
```

### Installed Packages

```coffee
# User
Sublime-Style-Column-Selection, v1.2.3
atom-color-highlight, v3.0.8
autocomplete-plus, v2.9.0
color-picker, v1.6.0
file-icons, v1.5.4
minimap, v4.7.6
minimap-find-and-replace, v4.2.0
save-session, v0.13.1
sublime-tabs, v0.5.3

# Dev
No dev packages
```

**Comment by xdumaine** [comment]
Yeah, I'm still getting this too.

**Comment by mpeterson2** [comment]
0.13.3 was just pushed with an error that caused this. It is now fixed in 0.13.4. I'm not sure if it completely solved your issue, but hopefully it helps.

**Comment by mpeterson2** [comment]
0.13.5 should also help with @tumri's issue as well.
